cachi2/core/package_managers/yarn/main.py

```diff
@@ -86,9 +89,13 @@ def _fetch_dependencies(source_dir: RootedPath, output_dir: RootedPath) -> None:
     :param output_dir: the directory where the yarn dependencies will be downloaded to.
     :raises CalledProcessError: if the 'yarn install' command fails.
     """
-    # the output_dir is where the globalFolder must be set to
-    # the yarn commands can be called by using the core.utils.run_cmd function
-    pass
+    cachi2_output = output_dir.join_within_root("deps", "yarnberry")
+
+    args = ["install", "--mode", "skip-build"]
+    try:
+        run_yarn_cmd(args, source_dir, {"YARN_GLOBAL_FOLDER": str(cachi2_output)})
+    except subprocess.CalledProcessError:
+        raise YarnBerryError(reason="Failed to fetch dependencies using 'yarn install'")
```

Comment:
IMO we should re-raise the CalledProcessError as YarnBerryError already in run_yarn_cmd

Otherwise every piece of code that uses run_yarn_cmd will have to do this